OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs

```diff
@@ -90,6 +90,10 @@ public ServerCreationLogic(Widget widget, ModData modData, Action onExit, Action
 			checkboxUPnP.OnClick = () => allowPortForward ^= true;
 			checkboxUPnP.IsDisabled = () => !Game.Settings.Server.AllowPortForward;
 
+			var labelUPnP = panel.GetOrNull<LabelWidget>("UPNP_NOTICE");
+			if (labelUPnP != null)
+				labelUPnP.IsVisible = () => !Game.Settings.Server.DiscoverNatDevices;
```

Comment:
On that note, `Game.Settings.Server.AllowPortForward` really needs to be moved from Settings to UPNP, because it has nothing to do with settings.